tfhe/src/high_level_api/tag.rs

```diff
@@ -0,0 +1,420 @@
+use crate::high_level_api::backward_compatibility::tag::TagVersions;
+use tfhe_versionable::{Unversionize, UnversionizeError, Versionize, VersionizeOwned};
+
+const STACK_ARRAY_SIZE: usize = std::mem::size_of::<Vec<u8>>() - 1;
+
+/// Simple short optimized vec, where if the data is small enough
```

Comment:
I feel like the SmallVec optimization might be premature and has other implications that cause more trouble at the moment than it solves

Reply:
What other implications ?

Reply:
the size, compatibility of compiler versions, the fact the default case copies an array of 0s around

Reply:
just like with just a Vec, the compiler would copy the bytes of the internal data (ptr, size, capacity) of the vec

Reply:
yep my bad

Reply:
The SmallVec size is 8 bytes (likely sizeof::<usize>) bigger than a plain Vec<T>

Reply:
I do agree that the SmallVec thing might be premature, but the complexity of it is not that high I think

Also, the vec of the Tag which only gets and return bytes or simple u64/u128 and the fact that the serialization/versioning returns bytes protects us in a sense.